api/core/tools/tool_file_manager.py

```diff
@@ -65,7 +65,7 @@ def create_file_by_raw(user_id: str, tenant_id: str,
         """
         extension = guess_extension(mimetype) or '.bin'
         unique_name = uuid4().hex
-        filename = f"/tools/{tenant_id}/{unique_name}{extension}"
```

Comment:
Hi, have you tested the difference of behavior among several storage? dose they have the same strategy of handling the first slash? we need to ensure that because we need to keep our system forwards compatible

Reply:
I have checked all places which invoked storage.save function, they all use the filepath without /

Reply:
yeah, but this path with "/" has used for months, lots of data has already been saved into it

Reply:
I just tested with s3 storage backend. it works without any issues.

Reply:
> yeah, but this path with "/" has used for months, lots of data has already been saved into it

its behavior should be same for the filename without / for Object storage like S3
for local storage, the code is as following. It should be same too. The // in file name will be treated as /

```
    def save(self, filename, data):
        if not self.folder or self.folder.endswith('/'):
            filename = self.folder + filename
        else:
            filename = self.folder + '/' + filename
```

Reply:
I'm checking it with other storage, thx

Reply:
Is `yarl` package a better way to assembly URLs as it does in #4224 ?

Reply:
checked, LGTM